### ISSUE TITLE
Cannot disable cell-var-from-loop on a splitted line when var was used before

### ISSUE THREAD

**Issue by ThomasFaivre** [PyCQA/pylint#3756]
Hello!

### Steps to reproduce
```
for i in range(5):
    i # /!\ This is important
    a = lambda p: p * \
        i # pylint: disable=cell-var-from-loop
```

### Current behavior
$ pip install git+https://github.com/PyCQA/pylint.git@master
$ pylint --version
pylint 2.6.0-dev1
astroid 2.4.2
Python 3.7.3 (default, Dec 20 2019, 18:57:59) 
[GCC 8.3.0]

$ pylint --disable=all --enable=cell-var-from-loop -s n toto.py
************* Module toto
toto.py:1:0: C0103: Constant name "a" doesn't conform to UPPER_CASE naming style (invalid-name)

Same with 2.5.3 and 2.5.0.

### Expected behavior
$ pip install pylint==2.4.4
$ pylint --version
pylint 2.4.4
astroid 2.3.3
Python 3.7.3 (default, Dec 20 2019, 18:57:59) 
[GCC 8.3.0]
$ pylint --disable=all --enable=cell-var-from-loop -s n toto.py
<Everything is OK>

**Comment by AWhetter** [comment]
Why would you refer to `i` in this way though? With all messages enabled, pylint will raise `pointless-statement`, which is correct.
Or is the point of the example to highlight that it is not possible to disable any message on line 4, not just `call-var-from-loop`?

**Comment by ThomasFaivre** [comment]
It's just to use the variable, because pylint does not complain if `i` is not used before. I can do the same with `print(i)` if you prefer, but that's is not really the issue here.

I am not sure that any message cannot be disabled, but I believe its the combination of for+lambda+backslash that causes issue with this specific message.

**Comment by ThomasFaivre** [comment]
Hello,

I'm digging this up because I noticed a change in behavior with a recent pylint version.
The `i` or `print(i)` is not required anymore to reproduce the issue that the `cell-var-from-loop` cannot be disabled from a splitted line:

```python
$ cat toto.py
for i in range(5):
    a = lambda p: p * \
        i # pylint: disable=cell-var-from-loop
$ pylint --version # according to pip, this is commit a6a544613fd6cd28c3213751173f1b0e4205d1c0
pylint 2.16.0-dev
astroid 2.12.13
Python 3.7.3 (default, Oct 31 2022, 14:04:00) 
[GCC 8.3.0]
$ pylint --disable=all --enable=cell-var-from-loop -s n toto.py
************* Module toto
toto.py:3:8: W0640: Cell variable i defined in loop (cell-var-from-loop)
```

Please note that I managed to disable the message with this:

```python
$ cat toto.py
for i in range(5):
    a = lambda p: p * \ # pylint: disable=cell-var-from-loop
        i
```

But that is a `SyntaxError`, so it is probably irrelevant ^^"

I have removed the problematic lines in my code because of the new `unnecessary-lambda-assignment` message. This made me want to test this issue with that message:

```python
$ cat toto.py
for i in range(5):
    a = lambda p: p * \
        i
$ pylint --disable=all --enable=unnecessary-lambda-assignment -s n toto.py
************* Module toto
toto.py:2:8: C3001: Lambda expression assigned to a variable. Define a function using the "def" keyword instead. (unnecessary-lambda-assignment)

$ cat toto.py
for i in range(5):
    a = lambda p: p * \
        i # pylint: disable=unnecessary-lambda-assignment
$ pylint --disable=all --enable=unnecessary-lambda-assignment -s n toto.py
```

So I was able to disable the `unnecessary-lambda-assignment` message from a splitted line, but not the `cell-var-from-loop` message.

I hope that will help the investigation!
